presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergOrcFileWriter.java

```diff
@@ -0,0 +1,293 @@
+        Map<Integer, Long> nullCounts = nullCountsBuilder.build();
+        Map<Integer, ByteBuffer> lowerBounds = lowerBoundsBuilder.build();
+        Map<Integer, ByteBuffer> upperBounds = upperBoundsBuilder.build();
+        // TODO: Add column size accounting to ORC column writers
+        return new Metrics(fileRowCount,
+                null,
+                valueCounts.isEmpty() ? null : valueCounts,
+                nullCounts.isEmpty() ? null : nullCounts,
+                lowerBounds.isEmpty() ? null : lowerBounds,
+                upperBounds.isEmpty() ? null : upperBounds);
+    }
+
+    private static Set<OrcColumnId> getExcludedColumns(ColumnMetadata<OrcType> orcColumns)
+    {
+        ImmutableSet.Builder<OrcColumnId> excludedColumns = ImmutableSet.builder();
+        populateExcludedColumns(orcColumns, ROOT_COLUMN, false, excludedColumns);
+        return excludedColumns.build();
+    }
+
+    private static void populateExcludedColumns(ColumnMetadata<OrcType> orcColumns, OrcColumnId orcColumnId, boolean shouldBeExcluded, ImmutableSet.Builder<OrcColumnId> excludedColumns)
+    {
+        if (shouldBeExcluded) {
+            excludedColumns.add(orcColumnId);
+        }
+        OrcType orcColumn = orcColumns.get(orcColumnId);
+        switch (orcColumn.getOrcTypeKind()) {
+            case LIST:
+            case MAP:
+                for (OrcColumnId child : orcColumn.getFieldTypeIndexes()) {
+                    populateExcludedColumns(orcColumns, child, true, excludedColumns);
+                }
+                return;
+            case STRUCT:
+                for (OrcColumnId child : orcColumn.getFieldTypeIndexes()) {
+                    populateExcludedColumns(orcColumns, child, shouldBeExcluded, excludedColumns);
+                }
+                return;
+            default:
+        }
+    }
+
+    private static int getIcebergId(OrcType orcColumn)
+    {
+        String icebergId = requireNonNull(orcColumn.getAttributes().get(ORC_ICEBERG_ID_KEY), format("ORC column %s doesn't have an associated Iceberg ID", orcColumn.toString()));
+        return Integer.parseInt(icebergId);
+    }
+
+    private static Optional<IcebergMinMax> toIcebergMinMax(ColumnStatistics orcColumnStats, org.apache.iceberg.types.Type icebergType)
+    {
+        IntegerStatistics integerStatistics = orcColumnStats.getIntegerStatistics();
+        if (integerStatistics != null) {
+            Object min = integerStatistics.getMin();
+            Object max = integerStatistics.getMax();
+            if (min == null || max == null) {
+                return Optional.empty();
+            }
+            if (icebergType.typeId() == org.apache.iceberg.types.Type.TypeID.INTEGER) {
+                min = toIntExact((Long) min);
+                max = toIntExact((Long) max);
+            }
+            return Optional.of(IcebergMinMax.builder(icebergType).setMin(min).setMax(max).build());
```

Comment:
I don't think we need a builder for this. All the usages are for all arguments. So this be
```java
return Optional.of(new IcebergMinMax(icebergType, min, max));
```
Which I find more readable here.